lib/src/main/java/io/ably/lib/types/MessageExtras.java

```diff
@@ -1,64 +1,131 @@
 package io.ably.lib.types;
 
 import com.google.gson.JsonElement;
-import io.ably.lib.util.Log;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
 import io.ably.lib.util.Serialisation;
-import org.msgpack.core.MessageFormat;
 import org.msgpack.core.MessagePacker;
 import org.msgpack.core.MessageUnpacker;
+import org.msgpack.value.ImmutableMapValue;
+import org.msgpack.value.ImmutableValue;
+import org.msgpack.value.Value;
+import org.msgpack.value.ValueFactory;
 
 import java.io.IOException;
+import java.lang.reflect.Type;
+import java.util.Map;
+import java.util.Objects;
 
 public final class MessageExtras {
 	private static final String TAG = MessageExtras.class.getName();
-	
-	// TODO what about the 1.1 push extension? (TM2i)
 
-	private final DeltaExtras delta; // may be null
-	
+	/**
+	 * Only intended for use with a MessageExtras instance received from Ably's servers (inbound).
+	 */
+	private final DeltaExtras delta;
+
+	/**
+	 * Only intended for use with MessageExtras instances created to be sent to Ably's servers (outbound).
```

Comment:
I'm not sure I understand the comment. Supporting `raw` is for content that isn't understood by this library, and I think that content can go in both directions.

Reply:
Fair enough. I think it's evolved since our discussion later yesterday. I shall refine this commentary.

Reply:
https://github.com/ably/ably-java/pull/581/commits/c56c859fc133f73c559fa7525d9e4409c95700e2